File: Java/Suma kolejnych liczb naturalnych.java
import java.util.List;
import java.util.Scanner;
import java.util.ArrayList;

public class Main 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        int k = input.nextInt();
        int n = input.nextInt();
        int x = (n - (k * (k - 1)) / 2) / k;
        List<Integer> result = new ArrayList<>();
        for (int i = 0; i < k; ++i)
            result.add(x + i);
        for (final int num : result)
            System.out.print(num + " ");
        input.close();
    }
}

File: C++/Suma kolejnych liczb naturalnych.cpp
#include <vector>
#include <iostream>

int main() 
{
    int k, n;
    std::cin >> k >> n;
    int x = (n - (k * (k - 1)) / 2) / k;
    std::vector<int> result;
    for (int i = 0; i < k; ++i)
        result.push_back(x + i);
    for (const int& num : result)
        std::cout << num << ' ';
    return 0;
}